cli/sub_command.go

```diff
@@ -77,12 +81,67 @@ func configureSubCommand(app commandHost) {
 	act.Flag("stream", "Subscribe to a specific stream (required JetStream)").PlaceHolder("STREAM").StringVar(&c.stream)
 	act.Flag("ignore-subject", "Subjects for which corresponding messages will be ignored and therefore not shown in the output").Short('I').PlaceHolder("SUBJECT").StringsVar(&c.ignoreSubjects)
 	act.Flag("wait", "Max time to wait before unsubscribing.").DurationVar(&c.wait)
+	act.Flag("report-subjects", "Subscribes to a subject pattern and builds a de-duplicated report of active subjects receiving data").UnNegatableBoolVar(&c.reportSubjects)
+	act.Flag("report-top", "Number of subjects to show when doing 'report-subjects'. Default is 10.").Default("10").IntVar(&c.reportSubjectsCount)
 }
 
 func init() {
 	registerCommand("sub", 17, configureSubCommand)
 }
 
+func startSubjectReporting(ctx context.Context, subjMu *sync.Mutex, subjectReportMap map[string]int64, subjectBytesReportMap map[string]int64, subjCount int) {
+	go func() {
+		ticker := time.NewTicker(time.Second)
+
+		for range ticker.C {
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+			default:
+				subjectRows := [][]any{}
+
+				if runtime.GOOS != "windows" {
+					fmt.Print("\033[2J")
+					fmt.Print("\033[H")
+				}
+
+				totalBytes := int64(0)
+				totalCount := int64(0)
+
+				subjMu.Lock()
+				keys := make([]string, 0, len(subjectReportMap))
+
+				for k := range subjectReportMap {
+					keys = append(keys, k)
+				}
+				// sort.Strings(keys)
+				// sort by count in descending order
+				sort.Slice(keys, func(i, j int) bool {
+					return subjectReportMap[keys[i]] > subjectReportMap[keys[j]]
+				})
+
+				for count, k := range keys {
+					subjectRows = append(subjectRows, []any{k, subjectReportMap[k], subjectBytesReportMap[k]})
```

Comment:
lets run subject count through humanize.Comma() and bytes through humanize.IBytes()